src/main/java/com/faforever/client/patch/GameUpdaterImpl.java

```diff
@@ -82,18 +83,22 @@ public GameUpdater addFeaturedModUpdater(FeaturedModUpdater featuredModUpdater)
         }).exceptionally(throwable -> {
           throwable = ConcurrentUtil.unwrapIfCompletionException(throwable);
           boolean allowReplaysWhileInGame = preferencesService.getPreferences().getForgedAlliance().isAllowReplaysWhileInGame();
-          if (throwable.getCause() instanceof AccessDeniedException && allowReplaysWhileInGame) {
+          if (allowReplaysWhileInGame) {
             log.info("Unable to update files and experimental replay feature is turned on " +
                 "that allows multiple game instances to run in parallel this is most likely the cause.");
-            throw new UnsupportedOperationException("Unable to patch Forged Alliance to the required version " +
-                "due to conflicting version running", throwable);
-          } else if (throwable instanceof UnsupportedOperationException) {
-            throw new CompletionException(throwable);
-          } else if (!allowReplaysWhileInGame) {
+            if (throwable.getCause() instanceof AccessDeniedException) {
+              throw new UnsupportedOperationException("Unable to patch Forged Alliance to the required version " +
+                  "due to conflicting version running", throwable);
+            } else if (throwable.getCause() instanceof ChecksumMismatchException) {
+              throw new CompletionException(throwable.getCause());
```

Comment:
Ah just realized this will only rethrow when the player has allowReplaysWhileinGame set which is probably not intended.

I can fix it though as there are a few things to clean up I have been meaning to do